Missing link to atomic on some architectures

On Debian, builds for libkiwix are failing on armel, mipsel, and a few other architectures because of a missing link to `atomic`. 

The error is something like: 
```c++  -o src/libkiwix.so.9.2.2 'src/25a6634@@kiwix@sha/meson-generated_.._.._static_kiwixlib-resources.cpp.o' 'src/25a6634@@kiwix@sha/book.cpp.o' 'src/25a6634@@kiwix@sha/bookmark.cpp.o' 'src/25a6634@@kiwix@sha/library.cpp.o' 'src/25a6634@@kiwix@sha/manager.cpp.o' 'src/25a6634@@kiwix@sha/libxml_dumper.cpp.o' 'src/25a6634@@kiwix@sha/opds_dumper.cpp.o' 'src/25a6634@@kiwix@sha/downloader.cpp.o' 'src/25a6634@@kiwix@sha/reader.cpp.o' 'src/25a6634@@kiwix@sha/entry.cpp.o' 'src/25a6634@@kiwix@sha/server.cpp.o' 'src/25a6634@@kiwix@sha/searcher.cpp.o' 'src/25a6634@@kiwix@sha/search_renderer.cpp.o' 'src/25a6634@@kiwix@sha/subprocess.cpp.o' 'src/25a6634@@kiwix@sha/aria2.cpp.o' 'src/25a6634@@kiwix@sha/tools_base64.cpp.o' 'src/25a6634@@kiwix@sha/tools_pathTools.cpp.o' 'src/25a6634@@kiwix@sha/tools_regexTools.cpp.o' 'src/25a6634@@kiwix@sha/tools_stringTools.cpp.o' 'src/25a6634@@kiwix@sha/tools_networkTools.cpp.o' 'src/25a6634@@kiwix@sha/tools_otherTools.cpp.o' 'src/25a6634@@kiwix@sha/kiwixserve.cpp.o' 'src/25a6634@@kiwix@sha/name_mapper.cpp.o' 'src/25a6634@@kiwix@sha/server_byte_range.cpp.o' 'src/25a6634@@kiwix@sha/server_etag.cpp.o' 'src/25a6634@@kiwix@sha/server_request_context.cpp.o' 'src/25a6634@@kiwix@sha/server_response.cpp.o' 'src/25a6634@@kiwix@sha/subprocess_unix.cpp.o' -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,--start-group -Wl,-soname,libkiwix.so.9 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wl,-z,relro -Wl,-z,now -pthread /usr/lib/arm-linux-gnueabi/libicui18n.so /usr/lib/arm-linux-gnueabi/libicuuc.so /usr/lib/arm-linux-gnueabi/libicudata.so /usr/lib/arm-linux-gnueabi/libzim.so /usr/lib/arm-linux-gnueabi/liblzma.so /usr/lib/arm-linux-gnueabi/libz.so /usr/lib/arm-linux-gnueabi/libzstd.so /usr/lib/arm-linux-gnueabi/libxapian.so /usr/lib/arm-linux-gnueabi/libpugixml.so /usr/lib/arm-linux-gnueabi/libcurl.so /usr/lib/arm-linux-gnueabi/libmicrohttpd.so -Wl,--end-group
/usr/bin/ld: src/25a6634@@kiwix@sha/server_request_context.cpp.o: in function `std::__atomic_base<unsigned long long>::fetch_add(unsigned long long, std::memory_order)':
/usr/include/c++/9/bits/atomic_base.h:541: undefined reference to `__atomic_fetch_add_8'
collect2: error: ld returned 1 exit status
```

Here are the full build logs: https://buildd.debian.org/status/package.php?p=libkiwix&suite=experimental

Searching around a bit, it seems we just need to add `-latomic` for those architectures. See e.g. how redis fixed this in https://github.com/antirez/redis/issues/6275 / https://github.com/antirez/redis/commit/f5d48537f1aa76e5ce4f14953517bd25c9ad4673

For meson I think we can use https://mesonbuild.com/Reference-manual.html#build_machine-object to vary what we link against.

--- a/meson.build
+++ b/meson.build
@@ -18,6 +18,11 @@ if 'java' in wrapper
   add_languages('java')
 endif
 
+# See https://github.com/kiwix/kiwix-lib/issues/371
+if target_machine.cpu_family() in ['arm', 'mips', 'm68k', 'ppc']
+  extra_libs += '-latomic'
+endif
+
 thread_dep = dependency('threads')
 libicu_dep = dependency('icu-i18n', static:static_deps)
 libzim_dep = dependency('libzim', version : '>=6.1.1', static:static_deps)

--- a/src/meson.build
+++ b/src/meson.build
@@ -53,6 +53,7 @@ kiwixlib = library('kiwix',
                    kiwix_sources,
                    include_directories : inc,
                    dependencies : all_deps,
+                   link_args: extra_libs,
                    version: meson.project_version(),
                    install: true,
                    install_dir: install_dir)

